### PR TITLE
fix(cmd): fix namespace representation in cmd output

### LINKED ISSUE #3335: Usability/DevUX Meta Issue
This issue collects smaller quirks with the current API/docs/CLI in one place. I will list them here and break them out into their own issues where it makes sense:

- [x] the comment here is wrong: https://github.com/celestiaorg/celestia-node/blob/f2b664f3abe5e76ed51e19305aa3489980a93692/blob/service.go#L113 (it only returns one blob; GetAll is what returns all blobs)
- [x] GetAll is confusing and edge-cases are under-specified: https://github.com/celestiaorg/celestia-node/issues/3292#issuecomment-2059313529
- [x] some examples in API docs are not representative; e.g. blob.Get returns a different commitment than requested (#3317 )
- [x] CLI: when looking at the for usage returned by `--help` encoding related guidance is often missing, eg. `celestia blob get [height] [namespace] [commitment] [flags]` should state which format the inputs are expected in (hex? base64?). The CLI help should be self-contained; currently one needs to e.g. look [here](https://docs.celestia.org/developers/node-tutorial) to know how to use this
- [x] CLI: namespaces are submitted as hex but returned base64 encoded (which is very confusing); example: 

``` 
celestia blob submit 0x42690c204d39600fddd3 0x676d                                                                                                                  
{
  "result": {
    "height": 1671475,
    "commitments": [
      "0MFhYKQUi2BU+U1jxPzG7QY2BVV1lb3kiU+zAK7nUiY="
    ]
  }
}
⋊> ~/G/celestia-node on main ⨯ celestia blob get-all 1671475  0x42690c204d39600fddd3                                                                                                               12:06:50
{
  "result": [
    {
      "namespace": "AAAAAAAAAAAAAAAAAAAAAAAAAEJpDCBNOWAP3dM=",
      "data": "0x676d",
      "share_version": 0,
      "commitment": "0MFhYKQUi2BU+U1jxPzG7QY2BVV1lb3kiU+zAK7nUiY=",
      "index": 531
    }
  ]
}
```

 - [x] CLI/API: the node daemon gets started with only one account and one can only submit PFBs from that single account; this does not reflect how people interact with chains using different wallets for different purposes (applies to e.g. `celestia blob submit` and a bunch of state related commands / rpcs) https://github.com/celestiaorg/celestia-node/issues/3295
 - [x] #3345
 - [ ] consumers of the API do not have sufficient control for retries on blob submission
 - [x] also related to the above: https://github.com/celestiaorg/celestia-node/issues/3164
 - [x] #3361
 - [x] API: Clarify `State.SubmitPayForBlob`  vs `Blob.Submit`: both methods are too similar and it's unclear when to use which (related: https://github.com/celestiaorg/celestia-node/pull/3349#issuecomment-2109995753)

### PR DESCRIPTION
**Pull request by vgonkivs**
Partly refs #3335

```
celestia blob get 1337536 0x42690c204d39600fddd3 IXg+08HV5RsPF3Lle8PH+B2TUGsGUsBiseflxh6wB5E=
{
  "result": {
    "namespace": "0x42690c204d39600fddd3",
    "data": "gm",
    "share_version": 0,
    "commitment": "IXg+08HV5RsPF3Lle8PH+B2TUGsGUsBiseflxh6wB5E=",
    "index": 1
  }
}
```
```
celestia blob get 1337536 0x42690c204d39600fddd3 IXg+08HV5RsPF3Lle8PH+B2TUGsGUsBiseflxh6wB5E= --base64=true
{
  "result": {
    "namespace": "AAAAAAAAAAAAAAAAAAAAAAAAAEJpDCBNOWAP3dM=",
    "data": "Z20=",
    "share_version": 0,
    "commitment": "IXg+08HV5RsPF3Lle8PH+B2TUGsGUsBiseflxh6wB5E=",
    "index": 1
  }
}
```

Current PR fixes namespace representation, so it will be consistent with the passed one. Blob cmds require hex-based namespaces only now.

### CODE REVIEW

**Review by ramin — APPROVED** [review]

File: nodebuilder/blob/cmd/blob.go [hunk 1]
```diff
@@ -76,6 +77,9 @@ var getCmd = &cobra.Command{
 			return fmt.Errorf("error parsing a height: %w", err)
 		}
 
+		if !strings.Contains(args[1], "0x") {
```
Comment (cristaloleg) [inline nodebuilder/blob/cmd/blob.go:80]: Nit(?): `strings.HasPrefix`

Comment (cristaloleg) [reply nodebuilder/blob/cmd/blob.go:80]: Same below

**Review by cristaloleg — APPROVED** [review]

File: nodebuilder/blob/cmd/blob.go (hunk 1 above)
Comment (vgonkivs) [reply nodebuilder/blob/cmd/blob.go:80]: true. thanks

**Review by ramin — APPROVED** [review]

**Comment by vgonkivs** [comment]
Yes, non-base64 looks weird and it does not work with get methods `!x\u003e\ufffd\ufffd\ufffd\ufffd\u001b\u000f\u0017r\ufffd{\ufffd\ufffd\ufffd\u001d\ufffdPk\u0006R\ufffdb\ufffd\ufffd\ufffd\ufffd\u001e\ufffd\u0007\ufffd`

**Review by walldiss — APPROVED** [review]

**Review by jcstein — APPROVED** [review]
utACK

**Pull request merged**